salt/utils/json.py

```diff
@@ -39,6 +39,7 @@ def find_json(raw):
     # Search for possible starts end ends of the json fragments
     for ind, _ in enumerate(lines):
         line = lines[ind].lstrip()
+        line = line[0] if line else line
```

Comment:
I'm bit confused with this part, maybe there are no such examples, but for me it looks like it can work with only pretty formated json. It's confusing because looks like it just takes only first symbol from the string with ignoring the rest part completely.

Reply:
It does work with pretty formatted JSONs only indeed. I don't think `json.find_json` supports unformatted JSONS - except for the best-effort search at the end, which just filters the text before it and then tries to load whatever is left.

Our implementation that you authored definitely doesn't support unformatted JSONs:

```
    for ind, _ in enumerate(lines):
        line = lines[ind].lstrip()
        line = line[0] if line else line
        if line == "{" or line == "[": # <-- this will never work for unformatted JSON
            starts.append((ind, line))
        if line == "}" or line == "]": # <-- this will never work for unformatted JSON
            ends.append((ind, line))
```